sandbox/libs/authn/src/main/java/org/opensearch/authn/PermissionFactory.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is used to create Permission instances. The factory can create both standard Permissions which have specific
+ * formatting requirements and legacy permissions which are not checked for validity on creation.
+ */
+public class PermissionFactory {
+
+    public final static String[] INVALID_CHARACTERS = new String[] { ":", "" }; // This is a placeholder for what may want to be banned
+
+    // This is a placeholder for the different actions which a permission may grant a Subject to perform
+    public final static String[] QUALIFIED_ACTIONS = new String[] { "CREATE", "READ", "WRITE", "DELETE", "UPDATE" };
+
+    // A placeholder for the different resources which a permission may grant a permission based on
+    public final static String[] QUALIFIED_RESOURCES = new String[] { "index", "indices", "cluster", "all" };
+
+    /**
+     * This function creates a standard permission instance. It includes checking that the permission that is being created
+     * is properly formatted.
+     */
+    public Permission createPermission(String permissionString) {
+
+        Permission newPermission = new Permission(permissionString);
+        if (permissionIsValidFormat(newPermission)) {
+            return newPermission;
+        }
+        throw new InvalidPermissionName(permissionString);
+    }
+
+    /**
+     * This function creates a permission without checking that the permission string is valid.
+     */
+    public Permission createLegacyPermission(String permissionString) {
+        return new Permission(permissionString);
+    }
+
+    /**
+     * Check that the permission does not contain any forbidden strings.
+     * Assumes that the permission is formatted as <resource>.<action>
+     */
+    public boolean permissionIsValidFormat(Permission permission) {
```

Comment:
Lets not return `boolean` and just throw `InvalidPermission` with a specific error message for the different scenarios.

Reply:
E.g. `Permission 'opensearch.index.update' is not a valid permission, it should contain a resource name or wildcard`

Reply:
That makes sense. I will make that change.